### PR TITLE
Use torchrun for dynamo/distributed.py

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89096
* #89015
* #88523
* __->__ #89149

Mainly wanted to confirm torchrun works fine with dynamo/ddp,
but it is also a better system than manually launching processes.

Partially addresses issue #1779

New run commands
------------

single process:
python benchmarks/dynamo/distributed.py [args]

multi-gpu (e.g. 2 gpu on one host):
torchrun --nproc_per_node 2 benchmarks/dynamo/distributed.py [args]

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: benchmarks/dynamo/dist_util.py [hunk 1]
```diff
@@ -25,9 +25,14 @@
 
 
 def setup(rank, world_size):
-    os.environ["MASTER_ADDR"] = "localhost"
-    os.environ["MASTER_PORT"] = "12355"
-    dist.init_process_group("nccl", rank=rank, world_size=world_size)
+    # set defaults in case torchrun isn't used; no idea why the if is needed, but it hangs torchrun otherwise
+    if not os.getenv("MASTER_ADDR"):
```
Comment (aazzolini) [inline benchmarks/dynamo/dist_util.py:29]: why do u need the check here but not for RANK and WORLD_SIZE?

**Review by aazzolini — APPROVED** [review]

File: benchmarks/dynamo/dist_util.py (hunk 1 above)
Comment (wconstab) [reply benchmarks/dynamo/dist_util.py:29]: i literally have no idea.  I think i shouldn't need the check, but without it torchrun was hanging on the call to init process group. i printed the env strings before/after my setting them and they were apparently the same..

File: benchmarks/dynamo/dist_util.py [hunk 2]
```diff
@@ -25,9 +25,14 @@
...
+        os.environ["MASTER_ADDR"] = os.getenv("MASTER_ADDR", "localhost")
+    if not os.getenv("MASTER_PORT"):
+        os.environ["MASTER_PORT"] = os.getenv("MASETER_PORT", "12355")
```
Comment (davidberard98) [inline benchmarks/dynamo/dist_util.py:32]: typo in the os.getenv "MASETER_PORT" ?

**Pull request closed without merging**